components/server/src/authorization/spicedb-authorizer.ts

```diff
@@ -5,88 +5,70 @@
-import { ResponseError } from "vscode-ws-jsonrpc";
 import {
     observespicedbClientLatency as observeSpicedbClientLatency,
     spicedbClientLatency,
 } from "../prometheus-metrics";
 import { SpiceDBClient } from "./spicedb";
-import { getExperimentsClientForBackend } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
-
-export type CheckResult = {
-    permitted: boolean;
-    err?: Error;
-    response?: v1.CheckPermissionResponse;
-};
-
-export const NotPermitted = { permitted: false };
 
 @injectable()
-export class Authorizer {
-    @inject(SpiceDBClient)
-    private client: SpiceDBClient;
+export class SpiceDBAuthorizer {
+    constructor(
+        @inject(SpiceDBClient)
+        private client: SpiceDBClient,
+    ) {}
 
     async check(
         req: v1.CheckPermissionRequest,
         experimentsFields?: {
             userID?: string;
             orgID?: string;
         },
-    ): Promise<CheckResult> {
+    ): Promise<boolean> {
         if (!this.client) {
-            return {
-                permitted: false,
-                err: new Error("Authorization client not available."),
-                response: v1.CheckPermissionResponse.create({}),
-            };
+            return false;
         }
 
         const featureEnabled = await getExperimentsClientForBackend().getValueAsync("centralizedPermissions", false, {
             user: { id: experimentsFields?.userID || "" },
             teamId: experimentsFields?.orgID,
         });
         if (!featureEnabled) {
-            return {
-                permitted: false,
-                err: new Error("Feature flag not enabled."),
-                response: v1.CheckPermissionResponse.create({}),
-            };
+            return false;
         }
 
         const timer = spicedbClientLatency.startTimer();
         try {
             const response = await this.client.checkPermission(req);
             const permitted = response.permissionship === v1.CheckPermissionResponse_Permissionship.HAS_PERMISSION;
-            const err = !permitted ? newUnathorizedError(req.resource!, req.permission, req.subject!) : undefined;
 
             observeSpicedbClientLatency("check", req.permission, undefined, timer());
 
-            return { permitted, response, err };
+            return permitted;
         } catch (err) {
+            // while in we're running two authorization systems in parallel, we do not hard fail on checks.
             log.error("[spicedb] Failed to perform authorization check.", err, { req });
             observeSpicedbClientLatency("check", req.permission, err, timer());
-
-            throw new AuthorizerError("Failed to perform authorization check", err);
+            return false;
         }
     }
 
     async writeRelationships(
         req: v1.WriteRelationshipsRequest,
         experimentsFields?: {
-            userID?: string;
-            teamID?: string;
+            orgID?: string;
```

Comment:
I removed the user check here because we should treat the org as the root for enabling/disabling permissions.